airflow/contrib/operators/dataproc_operator.py

```diff
@@ -205,10 +208,19 @@ def __init__(self,
         self.idle_delete_ttl = idle_delete_ttl
         self.auto_delete_time = auto_delete_time
         self.auto_delete_ttl = auto_delete_ttl
+        self.single_node = single_node
 
         assert not (self.custom_image and self.image_version), \
             "custom_image and image_version can't be both set"
 
+        assert (
+            self.single_node and
+            self.num_preemptible_workers + self.num_workers == 0
+        ) or (
+            not self.single_node and
+            self.num_preemptible_workers + self.num_workers != 0
+        )
```

Comment:
Maybe add a helpful message for the end user here?

Reply:
Changed behavior and added message when assertion fails